### PR TITLE
chore(version): Bump version

### PR DESCRIPTION
**Pull request by tyllark**
### Features
- feat(Auth): Add fetchCurrentDevice API ([#5251](https://github.com/aws-amplify/amplify-flutter/pull/5251))
- feat: bump json_annotation dependency to v4.9

### Fixes
- fix(datastore): Clear subscriptions on Stop ([#5253](https://github.com/aws-amplify/amplify-flutter/pull/5253))
- fix(datastore): Restart Sync Engine when network on/off ([#5218](https://github.com/aws-amplify/amplify-flutter/pull/5218))
- fix: push notification flush events ([#5215](https://github.com/aws-amplify/amplify-flutter/pull/5215))

### CODE REVIEW

**Review by khatruong2009 — APPROVED** [review]

File: packages/api/amplify_api/pubspec.yaml [hunk 1]
```diff
@@ -20,8 +20,8 @@ platforms:
 
 dependencies:
   amplify_api_dart: ">=0.5.3 <0.6.0"
```
Comment (Equartey) [inline packages/api/amplify_api/pubspec.yaml:22]: Min should be bumped: `amplify_api_dart: ">=0.5.4 <0.6.0"`

File: packages/auth/amplify_auth_cognito/pubspec.yaml [hunk 2]
```diff
@@ -19,11 +19,11 @@ platforms:
   web:
 
 dependencies:
-  amplify_analytics_pinpoint: ">=2.3.0 <2.4.0"
+  amplify_analytics_pinpoint: ">=2.4.0 <2.5.0"
   amplify_analytics_pinpoint_dart: ">=0.4.3 <0.5.0"
```
Comment (Equartey) [inline packages/auth/amplify_auth_cognito/pubspec.yaml:23]: `amplify_analytics_pinpoint_dart: ">=0.4.4 <0.5.0"`

File: packages/auth/amplify_auth_cognito/pubspec.yaml [hunk 3]
```diff
@@ -19,11 +19,11 @@ platforms:
...
-  amplify_auth_cognito_dart: ">=0.11.3 <0.12.0"
-  amplify_core: ">=2.3.0 <2.4.0"
-  amplify_flutter: ">=2.3.0 <2.4.0"
+  amplify_auth_cognito_dart: ">=0.11.4 <0.12.0"
+  amplify_core: ">=2.4.0 <2.5.0"
+  amplify_flutter: ">=2.4.0 <2.5.0"
   amplify_secure_storage: ">=0.5.4 <0.6.0"
```
Comment (Equartey) [inline packages/auth/amplify_auth_cognito/pubspec.yaml:27]: `amplify_secure_storage: ">=0.5.5 <0.6.0"`

File: packages/auth/amplify_auth_cognito_dart/pubspec.yaml [hunk 4]
```diff
@@ -10,11 +10,11 @@ environment:
 
 dependencies:
   amplify_analytics_pinpoint_dart: ">=0.4.3 <0.5.0"
```
Comment (Equartey) [inline packages/auth/amplify_auth_cognito_dart/pubspec.yaml:12]: `amplify_analytics_pinpoint_dart: ">=0.4.4 <0.5.0"`

File: packages/auth/amplify_auth_cognito_dart/pubspec.yaml [hunk 5]
```diff
@@ -10,11 +10,11 @@ environment:
...
-  amplify_core: ">=2.3.0 <2.4.0"
+  amplify_core: ">=2.4.0 <2.5.0"
   amplify_secure_storage_dart: ">=0.5.1 <0.6.0"
```
Comment (Equartey) [inline packages/auth/amplify_auth_cognito_dart/pubspec.yaml:14]: `amplify_secure_storage: ">=0.5.5 <0.6.0"`

File: packages/notifications/push/amplify_push_notifications/pubspec.yaml [hunk 6]
```diff
@@ -9,7 +9,7 @@ environment:
   flutter: ">=3.19.0"
 
 dependencies:
-  amplify_core: ">=2.3.0 <2.4.0"
+  amplify_core: ">=2.4.0 <2.5.0"
   amplify_secure_storage: ">=0.5.4 <0.6.0"
```
Comment (Equartey) [inline packages/notifications/push/amplify_push_notifications/pubspec.yaml:13]: `amplify_secure_storage: ">=0.5.5 <0.6.0"`

File: packages/notifications/push/amplify_push_notifications/pubspec.yaml [hunk 7]
```diff
@@ -23,7 +23,7 @@ dev_dependencies:
   amplify_secure_storage_dart: ">=0.5.1 <0.6.0"
```
Comment (Equartey) [inline packages/notifications/push/amplify_push_notifications/pubspec.yaml:23]: `amplify_secure_storage_dart: ">=0.5.2 <0.6.0"`

File: packages/notifications/push/amplify_push_notifications_pinpoint/pubspec.yaml [hunk 8]
```diff
@@ -14,12 +14,12 @@ platforms:
...
-  amplify_core: ">=2.3.0 <2.4.0"
-  amplify_flutter: ">=2.3.0 <2.4.0"
-  amplify_push_notifications: ">=2.3.0 <2.4.0"
+  amplify_analytics_pinpoint: ">=2.4.0 <2.5.0"
+  amplify_analytics_pinpoint_dart: ">=0.4.4 <0.5.0"
+  amplify_auth_cognito: ">=2.4.0 <2.5.0"
+  amplify_core: ">=2.4.0 <2.5.0"
+  amplify_flutter: ">=2.4.0 <2.5.0"
+  amplify_push_notifications: ">=2.4.0 <2.5.0"
   amplify_secure_storage: ">=0.5.4 <0.6.0"
```
Comment (Equartey) [inline packages/notifications/push/amplify_push_notifications_pinpoint/pubspec.yaml:23]: `amplify_secure_storage: ">=0.5.5 <0.6.0"`

File: packages/notifications/push/amplify_push_notifications_pinpoint/pubspec.yaml [hunk 9]
```diff
@@ -30,8 +30,8 @@ dev_dependencies:
   amplify_secure_storage_dart: ">=0.5.1 <0.6.0"
```
Comment (Equartey) [inline packages/notifications/push/amplify_push_notifications_pinpoint/pubspec.yaml:30]: `amplify_secure_storage_dart: ">=0.5.2 <0.6.0"`

File: packages/storage/amplify_storage_s3/pubspec.yaml [hunk 10]
```diff
@@ -19,10 +19,10 @@ platforms:
   web:
 
 dependencies:
-  amplify_core: ">=2.3.0 <2.4.0"
+  amplify_core: ">=2.4.0 <2.5.0"
   amplify_db_common: ">=0.4.3 <0.5.0"
```
Comment (Equartey) [inline packages/storage/amplify_storage_s3/pubspec.yaml:23]: `amplify_db_common: ">=0.4.5 <0.5.0"`

File: packages/storage/amplify_storage_s3_dart/pubspec.yaml [hunk 11]
```diff
@@ -9,20 +9,20 @@ environment:
   sdk: ^3.3.0
 
 dependencies:
-  amplify_core: ">=2.3.0 <2.4.0"
+  amplify_core: ">=2.4.0 <2.5.0"
   amplify_db_common_dart: ">=0.4.4 <0.5.0"
```
Comment (Equartey) [inline packages/storage/amplify_storage_s3_dart/pubspec.yaml:13]: `amplify_db_common: ">=0.4.5 <0.5.0"`

File: packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml [hunk 12]
```diff
@@ -9,20 +9,20 @@ environment:
   sdk: ^3.3.0
 
 dependencies:
-  amplify_core: ">=2.3.0 <2.4.0"
+  amplify_core: ">=2.4.0 <2.5.0"
   amplify_db_common_dart: ">=0.4.4 <0.5.0"
   amplify_secure_storage_dart: ">=0.5.1 <0.6.0"
```
Comment (Equartey) [inline packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml:14]: Min should be bumped: 
`amplify_db_common_dart: ">=0.4.5 <0.5.0"`
`amplify_secure_storage_dart: ">=0.5.2 <0.6.0"`

File: packages/common/amplify_db_common_dart/pubspec.yaml
Comment (Equartey) [inline packages/common/amplify_db_common_dart/pubspec.yaml:1]: This package needs to have its version bumped.

File: packages/worker_bee/worker_bee/pubspec.yaml
Comment (Equartey) [inline packages/worker_bee/worker_bee/pubspec.yaml:1]: Package version needs bump

**Review by Equartey — CHANGES_REQUESTED** [review]

File: packages/worker_bee/worker_bee/pubspec.yaml
Comment (tyllark) [reply packages/worker_bee/worker_bee/pubspec.yaml:1]: Updated!

File: packages/auth/amplify_auth_cognito/pubspec.yaml (hunk 3 above)
Comment (tyllark) [reply packages/auth/amplify_auth_cognito/pubspec.yaml]: Updated!

File: packages/auth/amplify_auth_cognito_dart/pubspec.yaml (hunk 4 above)
Comment (tyllark) [reply packages/auth/amplify_auth_cognito_dart/pubspec.yaml]: Updated!

File: packages/auth/amplify_auth_cognito_dart/pubspec.yaml (hunk 5 above)
Comment (tyllark) [reply packages/auth/amplify_auth_cognito_dart/pubspec.yaml]: Updated!

File: packages/notifications/push/amplify_push_notifications/pubspec.yaml (hunk 6 above)
Comment (tyllark) [reply packages/notifications/push/amplify_push_notifications/pubspec.yaml]: Updated!

File: packages/notifications/push/amplify_push_notifications/pubspec.yaml (hunk 7 above)
Comment (tyllark) [reply packages/notifications/push/amplify_push_notifications/pubspec.yaml]: Updated!

File: packages/notifications/push/amplify_push_notifications_pinpoint/pubspec.yaml (hunk 8 above)
Comment (tyllark) [reply packages/notifications/push/amplify_push_notifications_pinpoint/pubspec.yaml]: Updated!

File: packages/notifications/push/amplify_push_notifications_pinpoint/pubspec.yaml (hunk 9 above)
Comment (tyllark) [reply packages/notifications/push/amplify_push_notifications_pinpoint/pubspec.yaml]: Updated!

File: packages/storage/amplify_storage_s3/pubspec.yaml (hunk 10 above)
Comment (tyllark) [reply packages/storage/amplify_storage_s3/pubspec.yaml:23]: Updated!

File: packages/storage/amplify_storage_s3_dart/pubspec.yaml (hunk 11 above)
Comment (tyllark) [reply packages/storage/amplify_storage_s3_dart/pubspec.yaml]: Updated!

File: packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml (hunk 12 above)
Comment (tyllark) [reply packages/analytics/amplify_analytics_pinpoint_dart/pubspec.yaml]: Updated!

**Review by khatruong2009 — DISMISSED** [review]

**Review by khatruong2009 — APPROVED** [review]

**Review by khatruong2009 — DISMISSED** [review]

File: packages/amplify/amplify_flutter/CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,16 @@
+## 2.4.0
+
+### Features
+- feat(Auth): Add fetchCurrentDevice API ([#5251](https://github.com/aws-amplify/amplify-flutter/pull/5251))
+- feat: bump json_annotation dependency to v4.9
```
Comment (NikaHsn) [inline packages/amplify/amplify_flutter/CHANGELOG.md:5]: Q: why is this a feat?

File: packages/amplify_core/CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,13 @@
+## 2.4.0
+
+### Features
+- feat(Auth): Add fetchCurrentDevice API ([#5251](https://github.com/aws-amplify/amplify-flutter/pull/5251))
+- feat: bump json_annotation dependency to v4.9
```
Comment (NikaHsn) [inline packages/amplify_core/CHANGELOG.md:5]: same as above

File: packages/analytics/amplify_analytics_pinpoint/CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,8 @@
+## 2.4.0
+
+### Features
+- feat: bump json_annotation dependency to v4.9
```
Comment (NikaHsn) [inline packages/analytics/amplify_analytics_pinpoint/CHANGELOG.md:4]: same as a bove

File: packages/analytics/amplify_analytics_pinpoint_dart/CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,8 @@
+## 0.4.4
+
+### Fixes
+- fix: push notification flush events ([#5215](https://github.com/aws-amplify/amplify-flutter/pull/5215))
```
Comment (NikaHsn) [inline packages/analytics/amplify_analytics_pinpoint_dart/CHANGELOG.md:4]: should this be cleaned?

File: packages/api/amplify_api_dart/CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,11 @@
+## 0.5.4
+
+### Features
+- feat: bump json_annotation dependency to v4.9
```
Comment (NikaHsn) [inline packages/api/amplify_api_dart/CHANGELOG.md:4]: ditto

File: packages/auth/amplify_auth_cognito_dart/CHANGELOG.md [hunk 18]
```diff
@@ -1,3 +1,9 @@
+## 0.11.4
+
+### Features
+- feat(Auth): Add fetchCurrentDevice API ([#5251](https://github.com/aws-amplify/amplify-flutter/pull/5251))
+- feat: bump json_annotation dependency to v4.9
```
Comment (NikaHsn) [inline packages/auth/amplify_auth_cognito_dart/CHANGELOG.md:5]: ditto

File: packages/aws_common/CHANGELOG.md [hunk 19]
```diff
@@ -1,3 +1,11 @@
+## 0.7.2
+
+### Features
+- feat: bump json_annotation dependency to v4.9
```
Comment (NikaHsn) [inline packages/aws_common/CHANGELOG.md:4]: ditto

File: packages/aws_common/CHANGELOG.md [hunk 20]
```diff
@@ -1,3 +1,11 @@
...
+
+### Fixes
+- fix: push notification flush events ([#5215](https://github.com/aws-amplify/amplify-flutter/pull/5215))
```
Comment (NikaHsn) [inline packages/aws_common/CHANGELOG.md:7]: I think we should clean up this

File: packages/aws_signature_v4/CHANGELOG.md [hunk 21]
```diff
@@ -1,3 +1,8 @@
+## 0.6.2
+
+### Features
+- feat: bump json_annotation dependency to v4.9
```
Comment (NikaHsn) [inline packages/aws_signature_v4/CHANGELOG.md:4]: ditto

File: packages/notifications/push/amplify_push_notifications/CHANGELOG.md [hunk 22]
```diff
@@ -1,3 +1,7 @@
+## 2.4.0
+
+- Minor bug fixes and improvements
```
Comment (NikaHsn) [inline packages/notifications/push/amplify_push_notifications/CHANGELOG.md:3]: shall we add this fix to the change log?
- fix: push notification flush events ([#5215](https://github.com/aws-amplify/amplify-flutter/pull/5215))

File: packages/storage/amplify_storage_s3_dart/CHANGELOG.md [hunk 23]
```diff
@@ -1,3 +1,8 @@
+## 0.4.4
+
+### Features
+- feat: bump json_annotation dependency to v4.9
```
Comment (NikaHsn) [inline packages/storage/amplify_storage_s3_dart/CHANGELOG.md:4]: ditto

**Review by khatruong2009 — APPROVED** [review]

File: packages/common/amplify_db_common/pubspec.yaml
Comment (Equartey) [inline packages/common/amplify_db_common/pubspec.yaml:1]: This file needs a version bump, and packages that depend on this will need their min's bumped.

**Review by Equartey — APPROVED** [review]

**Pull request merged**